src/olympia/addons/models.py

```diff
@@ -1260,15 +1260,18 @@ def is_unreviewed(self):
     def is_public(self):
         return self.status == amo.STATUS_PUBLIC and not self.disabled_by_user
 
-    def has_complete_metadata(self):
-        return all(self.get_required_metadata())
-
-    def get_required_metadata(self):
-        if not self.has_listed_versions():
+    def has_complete_metadata(self, for_channel=None):
+        return all(self.get_required_metadata(for_channel=for_channel))
+
+    def get_required_metadata(self, for_channel=None):
+        if for_channel is None:
+            for_channel = (
+                amo.RELEASE_CHANNEL_LISTED if self.has_listed_versions()
+                else amo.RELEASE_CHANNEL_UNLISTED)
+        if for_channel == amo.RELEASE_CHANNEL_UNLISTED:
             # Add-ons with only unlisted versions have no required metadata.
             return []
-        latest_version = self.find_latest_version(
-            channel=amo.RELEASE_CHANNEL_LISTED)
+        latest_version = self.versions.latest()  # Version.from_upload does it.
```

Comment:
The addon is complete == it has all required metadata (there isn't a concept of being complete for a channel - addons don't exist in channels). 

`for_channel` is `imagine_if_this_addon_had_a_version_with_a_channel_of` would it be have all required metadata.  So, yeah, not straightforward - it's needed so we can know if an add-on will need extra metadata _before_ the version is actually added because we don't have a two stage submission for API uploads (unlike devhub where we add the version and then check if everything's there and force the developer to supply the information to submit the version for review).

I didn't really want to duplicate the logic for what is 'required metadata' but suggestions to make it less fragile/more understandable?